lib/cisco_node_utils/cmd_ref/fabricpath.yaml

```diff
@@ -0,0 +1,178 @@
+# fabricpath
+---
+# Fabricpath feature is not available on N3K and N9K
+_exclude: [/N3/, /N9K/]
+
+aggregate_multicast_routes:
+  _exclude:
+    - /N5K/
+  kind: boolean
+  config_get: "show run fabricpath"
+  config_get_token: '/^fabricpath multicast aggregate\-routes$/'
+  config_set: "<state> fabricpath multicast aggregate-routes"
+  default_value: false
+  
+allocate_delay:
+  kind: int
+  config_get: "show fabricpath timers"
+  config_get_token: '/^Allocate Delay Timer\s+:\s+(\d+)/'
+  config_set: "<state> fabricpath timers allocate-delay <delay>"
+  default_value: 10
+
+auto_switch_id: 
+  kind: int
+  config_get: "show fabricpath switch-id"
+  config_get_token: '/^\*\s+(\d+).*No$/'
+
+feature:
+  config_get: "show feature-set"
+  config_get_token: '/^fabricpath[\s]+[\d]+[\s]+(\S+)/'
+  config_set: "<state> feature-set fabricpath"
+
+feature_install:
+  config_set: "<state> install feature-set fabricpath"
+
+graceful_merge:
+  auto_default: false
+  kind: boolean
+  config_get: "show run fabricpath"
+  config_get_token: '/^fabricpath graceful\-merge disable$/'
+  config_set: "<state> fabricpath graceful-merge disable"
+  default_value: true
```

Comment:
The negating part is handled in the getter/setter code:

      graceful_merge_conf = config_get('fabricpath', 'graceful_merge')
      # opposite meaning with the cli
      return :enable if graceful_merge_conf.nil?
      graceful_merge_conf ? :disable : :enable

From: Chris Van Heuveln <notifications@github.com<mailto:notifications@github.com>>
Reply-To: cisco/cisco-network-node-utils <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Monday, November 23, 2015 at 1:30 PM
To: cisco/cisco-network-node-utils <cisco-network-node-utils@noreply.github.com<mailto:cisco-network-node-utils@noreply.github.com>>
Cc: Cisco Employee <dcherian@cisco.com<mailto:dcherian@cisco.com>>
Subject: Re: [cisco-network-node-utils] Collapse feature/fabricpath into develop (#102)


In lib/cisco_node_utils/cmd_ref/fabricpath.yaml<https://github.com/cisco/cisco-network-node-utils/pull/102#discussion_r45663483>:

> +
> +feature:
> +  config_get: "show feature-set"
> +  config_get_token: '/^fabricpath[\s]+[\d]+[\s]+(\S+)/'
> +  config_set: "<state> feature-set fabricpath"
> +
> +feature_install:
> +  config_set: "<state> install feature-set fabricpath"
> +
> +graceful_merge:
> +  auto_default: false
> +  kind: boolean
> +  config_get: "show run fabricpath"
> +  config_get_token: '/^fabricpath graceful\-merge disable$/'
> +  config_set: "<state> fabricpath graceful-merge disable"
> +  default_value: true


IMO this prop should either be renamed graceful_merge_disable or the default_value should be false...

Note that the getter returns true when it matches 'disable'... but the graceful-merge execution state is actually false in that case.

I'm looking at this from the view of graceful-merge state, not grace-merge disable state.

(config)# sh run fabricpath all | i graceful-merge
(config)#
(config)# fabricpath graceful-merge disable
(config)# sh run fabricpath all | i graceful-merge
fabricpath graceful-merge disable

(config)# no fabricpath graceful-merge disable
(config)# sh run fabricpath all | i graceful-merge
(config)#


-
Reply to this email directly or view it on GitHub<https://github.com/cisco/cisco-network-node-utils/pull/102/files#r45663483>.